kclvm/sema/src/advanced_resolver/mod.rs

```diff
@@ -93,9 +95,9 @@ impl<'ctx> Context<'ctx> {
 impl<'ctx> AdvancedResolver<'ctx> {
     pub fn resolve_program(
         program: &'ctx Program,
-        gs: GlobalState,
-        node_ty_map: NodeTyMap,
-    ) -> GlobalState {
+        gs: &'ctx mut GlobalState,
```

Comment:
It feels a bit strange here that all stateful state variables are not unified, some are `Arc<Mutex<T>>`, some are `Rc<RefCell<T>>`, and some are `&mut` borrowing. Should they all be unified as `Arc<Mutex<T>`, corresponding to the multi-threaded scenario of the language server.